lib/client/client.js

```diff
@@ -850,9 +850,9 @@ Client.prototype.connect = function connect () {
   }
 
   // Initialize socket events and LDAP parser.
-  function initSocket (url) {
+  function initSocket (server) {
```

Comment:
This is my gripe with the "no shadow" rule. This isn't really a shadowing since it is defining a new closure. And in this case, "server" isn't accurate I do not believe. Isn't this parameter literally an instance of `URL`?

Reply:
I called it `server` because that is what the variable is called that is passed in when this function is invoked. This is still shadowing because it is using a variable name from a parent scope.